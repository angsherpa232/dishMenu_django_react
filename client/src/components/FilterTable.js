import React, { useState } from "react";
import PropTypes from "prop-types";

import styles from "./FilterTable.module.css";

const FilterTable = ({ onClickHandler }) => {
  // State for managing the input field for searching dish
  const [dish, setDish] = useState("");

  const textChangeHandler = textValue => {
    setDish(textValue);
  };

  const searchBtnClick = event => {
    event.preventDefault();
    onClickHandler(dish);
  };

  return (
    <form className={styles.searchContainer}>
      <input
        className={styles.inputField}
        type="text"
        placeholder="Preferred dish"
        value={dish}
        onChange={event => textChangeHandler(event.target.value)}
      />
      <button className={styles.btn} onClick={searchBtnClick}>
        Search
      </button>
    </form>
  );
};

FilterTable.propTypes = {
  onClickHandler: PropTypes.func.isRequired
};

export default FilterTable;
