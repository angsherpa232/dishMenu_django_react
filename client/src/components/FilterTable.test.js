import React from "react";
import renderer from "react-test-renderer";
import FilterTable from "./FilterTable";

describe("FilterTable", () => {
  test("should match snapshot.", () => {
    const component = renderer.create(
      <FilterTable onClickHandler={() => {}} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
