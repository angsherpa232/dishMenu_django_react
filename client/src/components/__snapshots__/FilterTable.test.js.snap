// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTable should match snapshot. 1`] = `
<form
  className="searchContainer"
>
  <input
    className="inputField"
    onChange={[Function]}
    placeholder="Preferred dish"
    type="text"
    value=""
  />
  <button
    className="btn"
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
